algebird-core/src/main/scala/com/twitter/algebird/CountMinSketch.scala

```diff
@@ -436,7 +437,8 @@ case class CMSItem[K: Ordering](item: K, override val params: CMSParams[K]) exte
   override def ++(other: CMS[K]): CMS[K] = {
     other match {
       case other: CMSZero[_] => this
-      case other: CMSItem[K] => CMSInstance[K](params) + item + other.item
+      case other: CMSItem[K] => CMSItems(List(item, other.item), 2L, params)
```

Comment:
When `maxExactCount == 0` what should happen to this `case`?  That is, when it is zero my understanding is that the user wants to disable the exact counting optimization based on `CMSItems`, so this `case` statement should rather be "if (zero) oldCode else newCode" I think.

That being said, it looks to me as if we'd have to thread such if-else switches for maxExactCount through multiple parts of the CMS code, which is an indication that there might be a better pattern to enable/disable the new optimization functionality?